Add version 2018-11-27-preview to Microsoft.Insights

aydany opened the pull request:
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/aydany/azure-rest-api-specs/dev-monitor-Microsoft.Insights-2018-11-27-preview).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

aydany committed:
Adds base for updating Microsoft.Insights from version preview/2018-06-01-preview to version 2018-11-27-preview

aydany committed:
Updates readme

aydany committed:
Updates API version in new specs and examples

Commit:
Add initial swagger

Removed some of the autogenerated files and added the swagger file and
examples for the new api.

Commit:
Remove extra / from examples

Commit:
Fix swagger file

Commit:
Improve docs

Commit:
Update resource model

Made it the same as the rest of the specs

Commit:
Fix the resource data model: no location and tags

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-go

Commit:
Fix readme

aydany commented:
This changeset introduces a new extension resource in Microsoft.Insights. This resource is modelled as a singleton collection that provides information about the onboarding status of a resource/resource scope to VM Insights, a component of Azure Monitor.

The new resource has been reviewed with the ARM review board.

This API is not yet enabled in ARM. We were asked to submit this PR and have it approved, before making the manifest changes.

sarangan12 commented on specification/monitor/resource-manager/readme.md:
```diff
@@ -4,34 +4,61 @@
 description: Monitor Management Client
 openapi-type: arm
-
 tag: package-2018-09
```
Don't you want to add the new tag here?

aydany replied:
Initially, I opted not to, since this is a new preview version, but I made the change to generate the preview package by default. This exposed a couple of issues, which I resolved. One of them, however, appears to be a problem with the existing specs, and they will have to eventually be updated (and code changes made) to be compliant with the expected error response format.

Commit:
Generate the latest package

The latest package including support for the new API will be generated.
This required several changes:
- The defintion of Resource used by the other specs was incompatible.
This was because the other resources were tracked and the new ones is
proxy. Defined a new ProxyResource type.
- The ErrorResponse definition was incompatible with the rest of the
specs in the folder. This is actually a bug in the exesiting specs. They
do not respect the contract of  { "error": {"code": "", "message": ""}},
i.e., they lack the top-level error element. To work around this issue,
defined a new response type, ResponseWithError, which is compliant
with the latest ARM requirements.

ravbhatnagar commented:
This was reviewed in person by Vlad and @KrisBash. Kris - please feel free to sign off if it looks good.

sarangan12 commented:
Notified @lmazuel @anuchandy @daschult regarding the SDK generation failure and waiting for their update.

lmazuel commented:
@AutorestCI rebuild

AutorestCI commented:
<details><summary>Encountered an unknown error</summary><p>

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/github_tools.py", line 33, in exception_to_github
    yield context
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/bot_framework.py", line 105, in manage_comment
    response = getattr(self.handler, orderstr)(webhook_data.issue, *split_text)
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/restbot.py", line 36, in rebuild
    self.sdk_default_base
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/github_handler.py", line 151, in rest_pr_management
    sdk_tag=sdk_tag
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkNewCLI.py", line 306, in generate_sdk_from_git_object
    commit_sha = do_commit(sdk_repo, message, branch_name, commit_for_sha.sha)
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/git_tools.py", line 44, in do_commit
    msg = message_template.format(hexsha=hexsha)
KeyError: ' "error"'

```

</p></details>

AutorestCI commented:
<details><summary>Encountered an unknown error</summary><p>

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/github_tools.py", line 33, in exception_to_github
    yield context
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/bot_framework.py", line 105, in manage_comment
    response = getattr(self.handler, orderstr)(webhook_data.issue, *split_text)
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/restbot.py", line 36, in rebuild
    self.sdk_default_base
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/github_handler.py", line 151, in rest_pr_management
    sdk_tag=sdk_tag
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkNewCLI.py", line 306, in generate_sdk_from_git_object
    commit_sha = do_commit(sdk_repo, message, branch_name, commit_for_sha.sha)
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/git_tools.py", line 44, in do_commit
    msg = message_template.format(hexsha=hexsha)
KeyError: ' "error"'

```

</p></details>

AutorestCI commented:
<details><summary>Encountered an unknown error</summary><p>

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/github_tools.py", line 33, in exception_to_github
    yield context
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/bot_framework.py", line 105, in manage_comment
    response = getattr(self.handler, orderstr)(webhook_data.issue, *split_text)
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/restbot.py", line 36, in rebuild
    self.sdk_default_base
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/restapi/github_handler.py", line 151, in rest_pr_management
    sdk_tag=sdk_tag
  File "/usr/local/lib/python3.6/dist-packages/swaggertosdk/SwaggerToSdkNewCLI.py", line 306, in generate_sdk_from_git_object
    commit_sha = do_commit(sdk_repo, message, branch_name, commit_for_sha.sha)
  File "/usr/local/lib/python3.6/dist-packages/azure_devtools/ci_tools/git_tools.py", line 44, in do_commit
    msg = message_template.format(hexsha=hexsha)
KeyError: ' "error"'

```

</p></details>

lmazuel committed:
Update Python version

lmazuel commented:
@sarangan12 @aydany I added a commit to the PR to workaround a bug in our CI, because the last commit message contained a "{" character.

aydany commented:
@lmazuel Thank you!

aydany commented:
@sarangan12 What else do we need to resolve to complete this PR? Thanks!

KrisBash commented on specification/monitor/resource-manager/Microsoft.Insights/preview/2018-11-27-preview/vmInsightsOnboarding_API.json:
```diff
@@ -0,0 +1,260 @@
+                },
+                "description": "The onboarding status for the resource. Note that, a higher level scope, e.g., resource group or subscription, is considered onboarded if at least one resource under it is onboarded."
+              },
+              "dataStatus": {
```
So, dataStatus is an aggregate in the case of multiple destinations? I.e. if the VM were onboarded to multiple destinations, and only sending to one, it would still be present? Is this enough fidelity or should the status be a property of the destination?

KrisBash reviewed (approved):
Signing off with comment
